proposals/p1964.md

```diff
@@ -0,0 +1,275 @@
+representation of a single character's value within the source code of a
+computer program. Character literals between languages have some minor nuances
+but are fundamentally designed for the same purpose. Languages that have a
+dedicated character data type generally include character literals, for example
+C++, Java, Swift to name a few. Whereas other languages that lack distinct
+character type, like Python use strings of length one to serve the same purpose
+a character data type. For more information see
+[Character Literals Wiki](https://en.wikipedia.org/wiki/Character_literal),
+[Character Literals DBpedia](https://dbpedia.org/page/Character_literal)
+
+## Proposal
+
+Put character literals in single quotes, like `'a'`. Character literals work
+like numeric literals:
+
+-   Every different literal value has its own type.
+-   The bit width is determined by the type of the variable the literal is
+    assigned to, not the literal itself. Follows the plan from #1934.
+
+We will not support:
+
+-   Multi-line literals
+-   "raw" literals (using #'x'#)
+-   Empty character literals (`''`)
+-   ASCII Control codes (0...31), except for `\` when used for an escape
+    sequence. See
+    [String Literals: Escaping Sequence](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0199.md#escape-sequences-1)
+
+## Details
+
+A character literal is a sequence enclosed with single quotes delimiter ('),
+excluding:
+
+-   New line
+-   Single quote (`'`), but can write `'` character as `'\''`
+-   Back-slash (`\`), except when used to form an escape sequence. See
+    [String Literals: Escaping Sequence](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0199.md#escape-sequences-1)
+
+### Types
+
+Since we will be representing the character literal as a sequence consisting of
+elements that are each either a unicode code-point or a hex-encoded byte value.
+
+We can see whether the particular literal is represented in the variable's type
+by only looking at the types:
+
+```
+let allowed: u8 = 'a';
+```
+
+The above is allowed because the type of 'a' indicates that it can be converted
+to `u8`.
+
+```
+let error1: u8 = '😃';
+let error2: u8 = 'AB';
+```
+
+However these should produce errors, the types of '😃' and 'AB' indicate that
+they cannot be converted to the declared type `u8`. However any `'\n'` and
+`'\u{A}'` would be of the same type, as they are the same unicode entities
+`%0A`. It is worth noting that `'\x0A'` will be slightly different as any
+character following the `\x...` sequence representation are not the same as a
+Unicode character. For example:
+
+```
+var c1: EBCDICChar = '\x0A';
+var c2: EBCDICChar = '\n';
+```
+
+Here we assume that `c1` will be the EBCDIC RPT character (0xA) and for `c2` to
+be the EBCDIC NL character (0x15)
+
+### Operations
+
+From the previous examples, we should be able to use the following operators:
+
+-   Comparison: `<`, `>`, `<=`, `>=` `==`
```

Comment:
What do these operations mean when one operand is a code point like 'x' and the other is a code unit like `\x63'? Maybe we should reject?